src/fake-timers-src.js

```diff
@@ -740,7 +740,12 @@ function withGlobal(_global) {
         }
 
         if (typeof timer.func === "function") {
-            timer.func.apply(null, timer.args);
+            if (timer.animation) {
+                // For requestAnimationFrame, the arg is a reference to performance.now
+                timer.func.apply(null, [timer.args[0]()]);
```

Comment:
This code is kind of hairy without the comment.  I would like to clean it up somewhat.

The easiest option is probably just to extract this into a named function, like "handleRequestAnimationFrame" and inline the comment there, but it doesn't really remove the main issue, which is tightly coupling two separate bits of code (the setup and the execution of the timer). 

It would be better if we could handle this in a single branch.

What about simply moving this to the setup phase, wrapping it into a closure, so that this bit if the code will just be applying the argument to the function, no matter the type?

I am writing on my phone, so maybe o am missing something here, like args being bound too early, but conceptually that should be cleaner. Your tests should be unaffected by this.

Reply:
Yeah I didn't like this either... I don't think we can avoid passing the function around, although possibly it can be less magical and more consistent.

I'll try something in `addTimer` to normalize it and then you can tell me what you think.

Reply:
Oh here's an interesting idea, I just tried making `args` a getter for `requestAnimationFrame` and that works great. It creates the closure and the usage is identical since the consuming code cannot tell the difference between `timer.args` being an array or a getter which returns an array.

Hopefully it's not too "clever".